docs/proposal/kmesh-reliability.md

```diff
@@ -0,0 +1,164 @@
+- Authentication
+  - maps: map_of_tuple | map_of_auth
+  - Data source/Time: map_of_tuple is the table that reports the quintuple to be authenticated in the passive connect phase. The daemon authentication module updates the link information to be denied to map_of_auth in user mode. In the xdp packet receiving process, deny is performed based on map_of_auth.
+- Waypoint interconnection
+  - maps: map_of_dst_info
+  - Data source/Time: If ndat needs to be delivered to the waypoint in the active connect phase, the original link information is added to the map_of_dst_info, and the map_of_kmesh_socket sockhash table is added to replace the send/recv hook of the socket (to facilitate the BPF hook in the triggering of the receive and transmit processes). Reads the map_of_dst_info record from the BPF prog in the send phase, constructs meta data, and adds the data to the TCP packet.
+
+Impacts of the Kmesh restart on the preceding tables are as follows:
+
+- During the Kmesh restart:
+
+  | Scenario     | workload model | Kmesh management | Kmesh byPass | Auth | waypoint Interconnection                                     |
+  | ------------ | -------------- | ---------------- | ------------ | ---- | ------------------------------------------------------------ |
+  | Existed link | Y              | Y                | Y            | Y    | P(If the connection is complete but the send request has not been sent) |
+  | New link     | N              | N                | N            | N    | N                                                            |
+
+  Countermeasures:
+
+  - The bpf prog and map need to be maintained during restart.
+
+- After the Kmesh is restarted:
+
+  | Scenario     | workload model | Kmesh management | Kmesh byPass | Auth | waypoint Interconnection |
+  | ------------ | -------------- | ---------------- | ------------ | ---- | ------------------------ |
+  | Existed link | Y              | Y                | Y            | Y    | Y                        |
+  | New link     | Y              | N                | Y            | N    | N                        |
+
+  Countermeasures:
+
+  - The bpf prog and map need to be maintained during restart.
+  - After the Kmesh is started, the data in the map needs to be restored and the data needs to be synchronized with the control plane.
+
+  Note: Currently, Kmesh supports only link-level authentication.
+
+#### Solutions
+
+![](pics/reliability-arch.svg)
+
+- eBPF management
+
+  - To keep the governance capability during the Kmesh restart, you need to place the bpf prog and bpf map pin to the /sys/fs/bpf directory on the node.
+  - During the startup of the Kmesh, the BPF PROG is replaced and the BPF MAP is inherited.
+
+- Configure recovery
+
+  - Workload resources
+    - During the startup of the Kmesh, the local bpf map data is restored to the user-mode cache before the Kmesh subscribes to workload model data from the Istiod.
```

Comment:
Based on offline discussion, how can user space cache be restored correctly is still not clear